core/src/test/java/org/elasticsearch/index/fielddata/AbstractFieldDataTestCase.java

```diff
@@ -94,8 +95,8 @@ protected boolean hasDocValues() {
         } else if (type.getType().equals("byte")) {
             fieldType = MapperBuilders.byteField(fieldName).docValues(docValues).fieldDataSettings(type.getSettings()).build(context).fieldType();
         } else if (type.getType().equals("geo_point")) {
-            // norelease update to .before(Version.V_2_2_0 once GeoPointFieldV2 is fully merged
-            if (indexService.getIndexSettings().getIndexVersionCreated().onOrBefore(Version.CURRENT)) {
+            if (type.getSettings().getAsVersion(IndexMetaData.SETTING_VERSION_CREATED, Version.CURRENT).before(Version.V_2_2_0)
```

Comment:
They don't. But I agree they probably should. At the moment ```AbstractFieldDataTestCase.setup()``` creates the index settings which doesn't have any version randomization. When I added randomization it threw all sorts of backcompat exceptions in other test unrelated to this GeoPointField feature. To keep that change separate I went ahead and added the version randomization I need for GeoField backcompat testing in ```AbstractGeoFieldDataTestCase.getFieldDataSettings``` which has the side effect of seting the fieldType settings separately from the index settings.